saleor/product/utils/product_variants.py

```diff
@@ -0,0 +1,26 @@
+from django.core.exceptions import ValidationError
+
+from ..error_codes import ProductErrorCode
+
+
+def product_variant_exist(product):
+    if product.variants.first() is None:
```

Comment:
`exists()` is the fastest way to check the existence of objects in the database. But anyway, this function is also used in `products_variant_exists` and both `first()` and `exists()` will result in duplicated database queries.